forem/forem
# fetch preview card follow buttons in one network call

## What type of PR is this? (check all applicable)


## Description

When visiting a post with multiple comments, a follow status check was being sent for every follow button in a preview card. On posts with long discussions this is a substantial number of network requests.

This PR fixes that to use the bulk request. It also makes sure that when you submit a new comment, the only button that is fetched/manipulated is the newly added one.

NB: two requests can happen to the `bulk_show` endpoint when you view a post, because we initialize user follow buttons both from `initializeLocalStorageRender` and `initializePage`. 

## Related Tickets & Documents

Closes #14085

## QA Instructions, Screenshots, Recordings

- Visit a post with more than one comment
- Check the network tab, filtering by `/follows`. You should see a maximum of two network requests to the `bulk_show` endpoint
- Verify that the Follow buttons on the page work as expected - this includes: the right-hand sticky nav, the author byline profile preview card, and preview cards in comments
- Add a comment to the post
- Check that your newly added comment triggers a maximum of one more request to the follows API
- Check that your newly added comment has a profile preview card with "Edit profile" button, and that if you click on that button you are linked to edit your profile
- Have a click around other places follow buttons appear - e.g. user profile, own profile, podcasts, and have a quick check there's no regression

### UI accessibility concerns?

N/A

## Added/updated tests?

- [X] No, and this is why: Existing tests continue to verify that the cards are added and behave correctly

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

N/A

## [optional] What gif best describes this PR or how it makes you feel?

![Jessica Day awkward face](https://media.giphy.com/media/PccxLzGfUAw48/giphy.gif)
